Model/CatalogIngestion/ProductEventManager.php

```diff
@@ -193,6 +210,10 @@ public function sendProductEvent(ProductEventInterface $productEvent): bool
      */
     public function runInstantProductEvent(int $productId, string $type, int $websiteId = null): void
     {
+        if (in_array($productId, $this->alreadySentInstantProductIds)) {
```

Comment:
not it is just additional check to prevent double sending request in plugins stack, related to catalog inventory & legacy magento inventory flows. When catalog inventory doesn't have any source items its working as legacy and triggers update 2 times one for source items second for native catalog inventory.